pkg/controller/controller_instance_test.go

```diff
@@ -115,42 +115,34 @@ func TestReconcileInstanceWithAuthError(t *testing.T) {
 
 	testController.reconcileInstance(instance)
 
+	// verify that one catalog client action occurred
 	actions := fakeCatalogClient.Actions()
-	assertNumberOfActions(t, actions, 1)
-
-	updateAction := actions[0].(clientgotesting.UpdateAction)
-	if e, a := "update", updateAction.GetVerb(); e != a {
-		t.Fatalf("Unexpected verb on action; expected %v, got %v", e, a)
-	}
-	updateActionObject := updateAction.GetObject().(*v1alpha1.Instance)
-	if e, a := testInstanceName, updateActionObject.Name; e != a {
-		t.Fatalf("Unexpected name of instance created: expected %v, got %v", e, a)
-	}
-	if e, a := 1, len(updateActionObject.Status.Conditions); e != a {
-		t.Fatalf("Unexpected number of conditions: expected %v, got %v", e, a)
-	}
-	if e, a := "ErrorGettingAuthCredentials", updateActionObject.Status.Conditions[0].Reason; e != a {
-		t.Fatalf("Unexpected condition reason: expected %v, got %v", e, a)
+	if err := checkCatalogClientActions(actions, []catalogClientAction{
+		{
+			verb: "update",
+			checkObject: checkInstance(instanceDescription{
+				name:       testInstanceName,
+				conditions: []string{"ErrorGettingAuthCredentials"},
+			}),
+			getRuntimeObject: getRuntimeObjectFromUpdateAction,
+		},
+	}); err != nil {
+		t.Fatal(err)
 	}
 
 	// verify one kube action occurred
 	kubeActions := fakeKubeClient.Actions()
-	assertNumberOfActions(t, kubeActions, 1)
-
-	getAction := kubeActions[0].(clientgotesting.GetAction)
-	if e, a := "get", getAction.GetVerb(); e != a {
-		t.Fatalf("Unexpected verb on action; expected %v, got %v", e, a)
-	}
-	if e, a := "secrets", getAction.GetResource().Resource; e != a {
-		t.Fatalf("Unexpected resource on action; expected %v, got %v", e, a)
+	if err := checkKubeClientActions(kubeActions, []kubeClientAction{
```

Comment:
it would be pretty easy to add to this function, but I'd propose we do it in a follow-up so we can get this foundation. it would look like adding individual fields to `kubeClientAction` and then adding logic to `checkKubeClientActions` to check them against each action